py/builtinimport.c

```diff
@@ -232,13 +232,15 @@ STATIC void do_load(mp_module_context_t *module_obj, vstr_t *file) {
 
     #endif // MICROPY_MODULE_FROZEN
 
+    qstr file_qstr = qstr_from_str(file_str);
```

Comment:
For the case where it's importing a .mpy file, this now interns the filename when it didn't before.

Reply:
It always interns the filename as it sets the `__file__` attribute to the same value in `do_execute_raw_code`. This is true for all imports (.py, .mpy, frozen). I've updated the PR to make this more explicit (and save it having to search for the now-already-interned value).

Technically `__file__` might be disabled, but it is enabled at level=core.

Reply:
You're right, `do_execute_raw_code` is interning the filename... it just seems a bit off to me that you now need to pass a qstr as a filename everywhere (for lexing, etc) instead of a const-char-*.  But I guess that's the way everything is structured, and the reason for that is so that tracebacks can include the filename easily.